vizro-ai/examples/dashboard_ui/app.py

```diff
@@ -0,0 +1,191 @@
+from dash import Input, Output, State, callback, dcc
+from dash.exceptions import PreventUpdate
+from vizro import Vizro
+
+vm.Container.add_type("components", UserUpload)
+vm.Container.add_type("components", MyDropdown)
+vm.Container.add_type("components", OffCanvas)
+vm.Container.add_type("components", CodeClipboard)
+vm.Container.add_type("components", Icon)
+
+MyPage.add_type("components", UserPromptTextArea)
+MyPage.add_type("components", UserUpload)
+MyPage.add_type("components", MyDropdown)
+MyPage.add_type("components", OffCanvas)
+MyPage.add_type("components", CodeClipboard)
+MyPage.add_type("components", Icon)
+
+
+class CustomDashboard(vm.Dashboard):
+    """Custom Dashboard model."""
+
+    def build(self):
+        dashboard_build_obj = super().build()
+        dashboard_build_obj.children.append(dcc.Store(id="data-store-id", storage_type="local"))
+        dashboard_build_obj.children.append(dcc.Store(id="api-store-id", storage_type="local"))
+        dashboard_build_obj.children.append(dcc.Store(id="outputs-store-id", storage_type="session"))
+        return dashboard_build_obj
+
+
+SUPPORTED_MODELS = [
+    "gpt-4o-mini",
+    "gpt-4",
+    "gpt-4-turbo",
+    "gpt-3.5-turbo",
+    "gpt-4o",
+]
+
+
+plot_page = MyPage(
+    id="vizro_ai_plot_page",
+    title="Vizro AI - Plot",
+    layout=vm.Layout(
+        grid=[
+            [3, 3, -1, 5],
+            [1, 1, 2, 2],
+            [4, 4, 2, 2],
+            *[[0, 0, 2, 2]] * 6,
+        ]
+    ),
+    components=[
+        vm.Container(title="", components=[CodeClipboard(id="plot")]),
+        UserPromptTextArea(
+            id="text-area-id",
+        ),
+        vm.Graph(id="graph-id", figure=px.scatter(pd.DataFrame())),
+        vm.Container(
+            title="",
+            layout=vm.Layout(grid=[[1], [0]], row_gap="0px"),
+            components=[
+                UserUpload(
+                    id="data-upload-id",
+                    actions=[
+                        vm.Action(
+                            function=data_upload_action(),
+                            inputs=["data-upload-id.contents", "data-upload-id.filename"],
+                            outputs=["data-store-id.data"],
+                        ),
+                        vm.Action(
+                            function=display_filename(),
+                            inputs=["data-store-id.data"],
+                            outputs=["upload-message-id.children"],
+                        ),
+                    ],
+                ),
+                vm.Card(id="upload-message-id", text="Upload your data file (csv or excel)"),
+            ],
+        ),
+        vm.Container(
+            title="",
+            layout=vm.Layout(grid=[[0, 0, 1, 1, -1, -1, -1, 2, -1]], row_gap="0px", col_gap="4px"),
+            components=[
+                vm.Button(
+                    id="trigger-button-id",
```

Comment:
Shall we put "Run VizroAI" button on the right and model selection dropdown on the left?